### ISSUE TITLE
[Feature] HiddenField Rule - property ignoreWith

### ISSUE THREAD

**Issue by vanniktech** [checkstyle/checkstyle#2677]
In many builders you'll use `.with(varname)` and usually that one also does hide the field but since you anyway assign it, it does not really matter that much.

This is an addition for [Coding/HiddenField](http://checkstyle.sourceforge.net/config_coding.html#HiddenField).

There would be a new property for that named `ignoreWith` of type boolean with a default value of false.

Right now HiddenField correctly detects this:

```java
public Builder withFoo(@NonNull final Foo foo) {
    this.foo = foo;
    return this;
}
```

If however the property `ignoreWith` would be set to true, checkstyle would not raise the HiddenField warning in the above code.

All in all the new rule would work like `ignoreSetter` with the only difference that `withFoo` would be ignored rather than `setFoo`.

If you guys also like that property, I could work on that and submit a PR.

**Comment by romani** [comment]
http://checkstyle.sourceforge.net/report_issue.html#How_to_request_new_feature_for_existing_functionality

**Comment by vanniktech** [comment]
This is an addition for [Coding/HiddenField](http://checkstyle.sourceforge.net/config_coding.html#HiddenField).

There would be a new property for that named `ignoreWith` of type boolean with a default value of false.

Right now HiddenField correctly detects this:

```java
public Builder withFoo(@NonNull final Foo foo) {
    this.foo = foo;
    return this;
}
```

If however the property `ignoreWith` would be set to true, checkstyle would not raise the HiddenField warning in the above code.

All in all the new rule would work like `ignoreSetter` with the only difference that `withFoo` would be ignored rather than `setFoo`.

@romani Is that clear enough? If not please let me know.

**Comment by romani** [comment]
please read #619, your issue might be duplicate.

If it is a duplicate, please close your issue and switch to original issue.
Implementation might be very complicated, and there is no guaranty that checkstyle can reliably cover that cases. If you really want to contribute, please be welcome. As you show some results I will start helping you with design and code-review.

**Comment by vanniktech** [comment]
Yup this is a dupe of #619.
